public/components/datasources/components/manage/data_connection.tsx

```diff
@@ -294,6 +344,55 @@ export const DataConnection = (props: any) => {
     );
   };
 
+  useEffect(() => {
+    // Loads databases cache when cache is empty
+    if (datasourceDetails.name) {
+      const cache = CatalogCacheManager.getDataSourceCache().dataSources.find(
+        (datasource) => datasource.name === datasourceDetails.name
+      );
+      if (cache?.status !== 'Updated') {
+        startLoadingDatabases(datasourceDetails.name);
+        setIsRefreshing(true);
+      } else {
+        const cachedList: CachedDatabase[] =
+          CatalogCacheManager.getDataSourceCache().dataSources.find(
+            (cachedDataSource) => cachedDataSource.name === datasourceDetails.name
+          )?.databases || [];
+        if (cachedList) {
+          setCachedDatabases(cachedList);
+        }
+        setIsRefreshing(false);
+        setIsFirstTimeLoading(false);
+      }
+    }
+  }, [datasourceDetails.name]);
+
+  useEffect(() => {
+    // Handles loading tables and acceleration cache when cache is empty
+    const loadingStatuses = [DirectQueryLoadingStatus.RUNNING, DirectQueryLoadingStatus.WAITING];
+    setTablesIsLoading(loadingStatuses.includes(tablesLoadStatus));
+    setAccelerationsIsLoading(loadingStatuses.includes(accelerationsLoadStatus));
+    if (databasesLoadStatus === DirectQueryLoadingStatus.SUCCESS) {
```

Comment:
I feel this piece of code are doing some unnecessary component updates, can we consider consolidate some of the state like @ps48 mentioned above to something like:
```
const [state, setState] = useState({
  cachedDatabases: [],
  cachedTables: [],
  cachedAccelerations: [],
  isRefreshing: true,
  isFirstTimeLoading: true,
});

// update once for all
setState(prevState => ({
  ...prevState,
  cachedDatabases: databases,
  isRefreshing: false,
  isFirstTimeLoading: false,
}));
``` 

Is it possible to consolidate the status updates for all components and apply them in a single operation?